packages/mui-base/src/AutocompleteUnstyled/useAutocomplete.js

```diff
@@ -1110,11 +1111,37 @@ export default function useAutocomplete(props) {
       tabIndex: -1,
       ...(!readOnly && { onDelete: handleTagDelete(index) }),
     }),
-    getListboxProps: () => ({
+    getListboxProps: ({ onScroll } = {}) => ({
       role: 'listbox',
       id: `${id}-listbox`,
       'aria-labelledby': `${id}-label`,
       ref: handleListboxRef,
+      onScroll: (event) => {
+        if (onScrollToBottom) {
+          const listboxNode = listboxRef.current;
+          const listboxHeight = listboxNode.offsetHeight;
+          const contentHeight = listboxNode.scrollHeight;
+          const scrollPosition = listboxNode.scrollTop;
+
+          const listboxPaddingBottom = Number(
+            window
+              .getComputedStyle(listboxNode)
+              .getPropertyValue('padding-bottom')
+              .replace('px', ''),
+          );
+
+          const distanceFromBottom =
+            contentHeight - (scrollPosition + listboxHeight + listboxPaddingBottom);
+          //   In general, it's best to use a small threshold value when checking for the scroll position, since the scrollTop, offsetHeight, and scrollHeight values are often not perfectly accurate. For example, some browsers might round these values to the nearest integer, which can cause small discrepancies.
+          if (distanceFromBottom <= 1 && listboxHeight !== contentHeight) {
+            // The scrollbar is at the bottom of the listbox element
+            onScrollToBottom(event);
+          }
+        }
```

Comment:
I've actually used most of the logic from that comment except adding of padding bottom. Explaining reason for adding `padding-bottom` is bit difficult but anyway i'll try.

Here `x` is listbox.scrollTop and `y` is `listbox.offsetHeight` , `x+y+z = listbox.scrollHeight` . To know whether scrollBar has reached bottom or not we have to observe `z` value, if `z` is `0` then scroll bar has reached bottom.

But according to current implementation `z` value can never become `0` because `padding-bottom` is not added to `scrollTop` (https://github.com/mui/material-ui/blob/bf42a54f1eba258542d5b7df391efa468b6a39a8/packages/mui-base/src/AutocompleteUnstyled/useAutocomplete.js#L382-L393), logic only considers option's height while calculating scrollTop, so a part of listbox remains invisble when navigating through keyboard. (you can refer this demo to get more understanding of what i'm speaking (https://stackblitz.com/edit/react-4eqnxs?file=demo.js). Notice padding-bottom is set to 5 rem => navigate to last option through `keyboard` and again to `1st option` through `keyboard` => notice how bottom part of listbox remain invisble)

here `z` can be further divided in to parts `z = padding-bottom + height of invisble options`, If `padding-bottom + x + y = listbox.scrollHeight` then scollBar has reached bottom.  So that was the reason i've substracted `paddingBottom` value.


<img width="817" alt="Screenshot 2023-02-07 at 11 14 58 PM" src="https://user-images.githubusercontent.com/60743144/217323791-32c4ff27-edbb-4ae2-8375-495b0e3b4b7a.png">

Reply:
But doesn't the `listboxHeight` i.e `offsetHeight` already include the listbox padding?

According to the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetHeight) for offsetHeight,
> The HTMLElement.offsetHeight read-only property returns the height of an element, including vertical padding and borders, as an integer.

Reply:
@sai6855 Can you address the above comment?

Reply:
Since it already contains `padding`, did you try without adding `paddingBottom`?

Reply:
I tried the following based on the example shown in [the documentation](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollHeight#problems_and_solutions) (Determine if an element has been totally scrolled) and it works well:
```diff
--- a/packages/mui-base/src/useAutocomplete/useAutocomplete.js
+++ b/packages/mui-base/src/useAutocomplete/useAutocomplete.js
@@ -1132,22 +1132,10 @@ export default function useAutocomplete(props) {
       onScroll: (event) => {
         if (onScrollToBottom) {
           const listboxNode = listboxRef.current;
-          const listboxHeight = listboxNode.offsetHeight;
-          const contentHeight = listboxNode.scrollHeight;
-          const scrollPosition = listboxNode.scrollTop;
-
-          const listboxPaddingBottom = Number(
-            window
-              .getComputedStyle(listboxNode)
-              .getPropertyValue('padding-bottom')
-              .replace('px', ''),
-          );
-
-          const distanceFromBottom =
-            contentHeight - (scrollPosition + listboxHeight + listboxPaddingBottom);
-          //   In general, it's best to use a small threshold value when checking for the scroll position, since the scrollTop, offsetHeight, and scrollHeight values are often not perfectly accurate. For example, some browsers might round these values to the nearest integer, which can cause small discrepancies.
-          if (distanceFromBottom <= 1 && listboxHeight !== contentHeight) {
-            // The scrollbar is at the bottom of the listbox element
+          if (
+            Math.abs(listboxNode.scrollHeight - listboxNode.clientHeight - listboxNode.scrollTop) <
+            1
+          ) {
             onScrollToBottom(event);
           }
         }
```

Reply:
But the keyboard navigation isn't working as expected with the above solution. You can keep yours if that's the case.